docs/deep-dives/pnpm.md

```diff
@@ -0,0 +1,341 @@
+abstracting the dependency management of those packages across SEEK projects.
+When you are using **skuba**,
+you do not need to declare these packages as direct `devDependencies`.
+In our previously-recommended package manager, [Yarn], these packages and others are automatically hoisted to create a flattened dependency tree.
+
+```json
+{
+  "devDependencies": {
+    "skuba": "7.2.0"
+  }
+}
+```
+
+```console
+node_modules
+├── jest
+├── prettier
+├── skuba
+└── other-skuba-deps
+```
+
+However, this behaviour can lead to some [silly bugs] when updating packages.
+
+### pnpm in skuba
+
+[pnpm] addresses the hoisting issue with a [symlinked structure].
+Each package is guaranteed to resolve compatible versions of its dependencies, rather than whichever versions were incidentally hoisted.
+
+This behaviour is a double-edged sword for a toolkit like **skuba**.
+Dependencies like Prettier and ESLint end up nested in a `node_modules/skuba/node_modules` subdirectory,
+where most editor and developer tooling integrations will not know to look.
+
+```console
+node_modules
+├── skuba -> ./.pnpm/skuba@7.2.0
+└── .pnpm
+    ├── skuba@7.2.0
+    │   └── node_modules
+    │       └── prettier -> ../../prettier@3.0.0
+    └── prettier@3.0.0
+        └── node_modules
+            └── other-dep -> <store>/other-dep
+```
+
+### .npmrc
+
+pnpm allows us to specify dependencies to hoist via command line or [`.npmrc`].
+The number of package patterns we need to hoist may fluctuate over time,
+so specifying hoist patterns via command line would be difficult to maintain.
+
+The **skuba**-maintained `.npmrc` currently instructs pnpm to hoist the following dependencies:
+
+```shell
+# managed by skuba
+public-hoist-pattern[]="@types*"
+public-hoist-pattern[]="*eslint*"
+public-hoist-pattern[]="*prettier*"
+public-hoist-pattern[]="esbuild"
+public-hoist-pattern[]="jest"
+public-hoist-pattern[]="tsconfig-seek"
+# end managed by skuba
+```
+
+From the previous example, this will produce the following `node_modules` layout,
+allowing external integrations to find `prettier` in `node_modules/prettier` as before.
+
+```console
+node_modules
+├── prettier -> ./.pnpm/prettier@3.0.0
+├── skuba -> ./.pnpm/skuba@7.2.0
+└── .pnpm
+    ├── skuba@7.2.0
+    │   └── node_modules
+    │       └── prettier -> ../../prettier@3.0.0
+    └── prettier@3.0.0
+        └── node_modules
+            └── other-dep -> <store>/other-dep
+```
+
+Committing pnpm configuration in `.npmrc` can conflict with build pipelines that synthesise an ephemeral `.npmrc` to access private SEEK packages on the npm registry.
+A solution to this problem is detailed in the migration guide below.
+
+---
+
+## Migrating from Yarn 1.x to pnpm
+
+This migration guide assumes that your project was scaffolded with a **skuba** template.
+
+1. Install **skuba** 7.4.0 or greater
+
+2. Install pnpm
+
+   Run the following on macOS:
+
+   ```bash
+   brew install pnpm
```

Comment:
I've been using `corepack enable` https://pnpm.io/installation#using-corepack

Thoughts?

Reply:
You can use in docker images too

Reply:
Not ignoring the feedback, but we might workshop this in another PR if sam's happy with the feedback as I want to get this out as a first pass alongside the release

Reply:
Did you have `corepack` on your computer by default?

I was using `volta` and `fnm` and it didn't seem to install it for me by default, so I figured instead of `brew install corepack` followed by `corepack enable pnpm`, `brew install pnpm` would be easier?

Reply:
Sorry, yeah I'm pretty sure I had corepack already. It's not in my `brew ls`. I use `nvm` as my node manager and corepack has been in node since node 16 i think.